iml-agent/src/agent_error.rs

```diff
@@ -71,11 +71,28 @@ impl std::error::Error for CibError {
     }
 }
 
+#[derive(Debug)]
+pub struct InterfaceError(pub String);
```

Comment:
Why do we need a new Error struct here when the contents is just a String?

Reply:
I thought it looked a little cleaner but I can just make it a string if you prefer.

Reply:
I don't think we need the extra level of indirection since it's all in the same crate.